File: aspnet5-angular2-typescript/src/PhotoGallery/Entities/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authentication;

namespace PhotoGallery.Entities
{
    public class Message : IEntityBase
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public string Date { get; set; }
        public int SenderId { get; set; }
        public int GroupId { get; set; }

        public virtual User User { get; set; }
        public virtual Group Group { get; set; }

        public override bool Equals(object obj)
        {
            var message = obj as Message;
            return message != null && Id == message.Id && Text == message.Text && Date == message.Date &&
                   SenderId == message.SenderId && GroupId == message.GroupId;
        }
    }
}

File: aspnet5-angular2-typescript/src/PhotoGallery/wwwroot/app/core/domain/message.ts
export class  Message {
    Id:number;
    ChatId:number;
    Date: string;
    SenderId:number;
    Text:string;
    constructor(id:number, chatId:number, date:string, senderId:number, text:string) {
        this.Id=id;
        this.ChatId=chatId;
        this.Date=date;
        this.SenderId=senderId;
        this.Text=text
    }
}